src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -300,13 +300,13 @@ class Direct$Type$Buffer$RW$$BO$
 #if[rw]
 
     public long address() {
-        ResourceScope scope = scope();
+        MemorySession scope = session();
         if (scope != null) {
-            if (scope.ownerThread() == null) {
-                throw new UnsupportedOperationException("ByteBuffer derived from shared segments not supported");
+            if (scope.ownerThread() == null && scope.isCloseable()) {
+                throw new UnsupportedOperationException("ByteBuffer derived from closeable shared segments not supported");
```

Comment:
Reviewer note: does user have the power to create non-closeable shared segments now?